File: go/leap/leap.go
// Package leap provies a leap year detection utility.
package leap

const testVersion = 2

// IsLeapYear takes a year and returns if it is a leap year.
func IsLeapYear(year int) bool {
	if year%400 == 0 {
		// 400 is a product of 4 and 100 - if divisible all conditions are met
		return true
	}
	if year%100 == 0 {
		// if the year was not divisible by 4, it would've been rejected anyway
		return false
	}
	if year%4 == 0 {
		return true
	}
	return false
}

File: python/leap/leap.py
#
# leap.py by Jack Hayhurst for exercism.io
# returns whether the given year is a leap year

def is_leap_year(year):
    """ return if the given year is a leap year"""
    if year % 400 is 0:
        return True
    if (year % 4 is 0) and (year % 100 is not 0):
        return True
    return False